tests/commonTemplates_test.go

```diff
@@ -0,0 +1,155 @@
+package tests
+
+import (
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	templatev1 "github.com/openshift/api/template/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	commonTemplates "kubevirt.io/ssp-operator/internal/operands/common-templates"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+var (
+	viewRole = &testResource{
+		Name:       commonTemplates.ViewRoleName,
+		Namsespace: commonTemplates.GoldenImagesNSname,
+		resource:   &rbac.Role{},
+	}
+	viewRoleBinding = &testResource{
+		Name:       commonTemplates.ViewRoleName,
+		Namsespace: commonTemplates.GoldenImagesNSname,
+		resource:   &rbac.RoleBinding{},
+	}
+	editClusterRole = &testResource{
+		Name:     commonTemplates.EditClusterRoleName,
+		resource: &rbac.ClusterRole{},
+	}
+	goldenImageNS = &testResource{
+		Name:     commonTemplates.GoldenImagesNSname,
+		resource: &core.Namespace{},
+	}
+	testTemplate = &testResource{
+		Name:       "centos6-server-large-v0.11.3",
+		Namsespace: commonTemplatesTestNS,
+		resource:   &templatev1.Template{},
+	}
+)
+
+var _ = Describe("Common templates", func() {
+	Context("resource creation", func() {
+		table.DescribeTable("created cluster resource", func(res *testResource) {
+			resource := res.NewResource()
+			err := apiClient.Get(ctx, client.ObjectKey{Name: res.Name}, resource)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hasOwnerAnnotations(resource.GetAnnotations())).To(BeTrue())
+		},
+			table.Entry("edit role", editClusterRole),
+			table.Entry("golden images namespace", goldenImageNS),
+		)
+
+		table.DescribeTable("created namespaced resource", func(res *testResource) {
+			err := apiClient.Get(ctx, client.ObjectKey{
+				Name: res.Name, Namespace: res.Namsespace,
+			}, res.NewResource())
+			Expect(err).ToNot(HaveOccurred())
+		},
+			table.Entry("view role", viewRole),
+			table.Entry("view role binding", viewRoleBinding),
+		)
+
+		It("Create common-template in custom NS", func() {
+			err := apiClient.Get(ctx, client.ObjectKey{
+				Name: testTemplate.Name, Namespace: commonTemplatesTestNS,
+			}, testTemplate.NewResource())
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("resource deletion", func() {
+		table.DescribeTable("recreate after delete", func(res *testResource) {
+			resource := res.NewResource()
+			resource.SetName(res.Name)
+			resource.SetNamespace(res.Namsespace)
+			Expect(apiClient.Delete(ctx, resource)).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				return apiClient.Get(ctx, client.ObjectKey{
+					Name: res.Name, Namespace: res.Namsespace,
+				}, resource)
+			}, timeout, time.Second).ShouldNot(HaveOccurred())
+		},
+			table.Entry("view role", viewRole),
+			table.Entry("view role binding", viewRoleBinding),
+			table.Entry("testTemplate in custom NS", testTemplate),
+		)
+		table.DescribeTable("recreate after delete", func(res *testResource) {
```

Comment:
These two tables called "recreate after delete" can be merged to one table. The code is the same except for assigning namespace in the ObjectKey. For cluster resources, the namespace is `""` so it can be assigned to ObjectKey.

Reply:
Merged now